OpenRA.Mods.Common/Traits/Upgrades/DeployToUpgrade.cs

```diff
@@ -20,9 +19,13 @@ namespace OpenRA.Mods.Common.Traits
 {
 	public class DeployToUpgradeInfo : ITraitInfo, Requires<UpgradeManagerInfo>
 	{
+		[UpgradeGrantedReference]
+		[Desc("The upgrades to grant while the actor is undeployed.")]
+		public readonly string[] UndeployedUpgrades = { };
+
 		[UpgradeGrantedReference, FieldLoader.Require]
-		[Desc("The upgrades to grant when deploying and revoke when undeploying.")]
-		public readonly string[] Upgrades = { };
+		[Desc("The upgrades to grant after deploying and revoke before undeploying.")]
+		public readonly string[] DeployedUpgrades = { };
```

Comment:
> [21:32] <+penev> abcdefg30: basically I'm not gonna do a half-assed upgrade rule (because it won't be able to do the job fully anyway) for the *one* use-case in Nolt's mod, which would even get merged into prep